app/uk/gov/hmrc/transitmovements/services/XmlParsers.scala

```diff
@@ -50,14 +50,12 @@ object XmlParsers extends XmlParsingServiceHelpers {
       case exception: DateTimeParseException => Left(ParseError.BadDateTime("preparationDateAndTime", exception))
     }
 
-  val messageTypeExtractor: Flow[ParseEvent, ParseResult[MessageType], NotUsed] = Flow[ParseEvent]
-    .mapConcat {
-      case s: StartElement if MessageType.values.exists(_.rootNode == s.localName) =>
-        Seq(MessageType.values.find(_.rootNode == s.localName).get)
-      case _ => Seq.empty
-    }
-    .take(1)
-    .fold[Either[ParseError, MessageType]](Left(ParseError.InvalidMessageType()))(
-      (_, next) => Right(next)
-    )
+  def movementReferenceNumberExtractor: Flow[ParseEvent, ParseResult[MovementReferenceNumber], NotUsed] =
+    XmlParsing
+      .subtree("CC028C" :: "TransitOperation" :: "MRN" :: Nil)
+      .collect {
+        case element if element.getTextContent.nonEmpty => MovementReferenceNumber(element.getTextContent)
+      }
+      .single("MRN")
```

Comment:
i.e., rather than using `single` here, can we let something go through if it's zero or one, rather than just one?